fastavro/_reader.pyx

```diff
@@ -18,14 +18,14 @@ except ImportError:
     import json
 
 from ._six import (
-    MemoryIO, xrange, btou, iteritems, is_str, str2ints, fstint
+    btou, iteritems, is_str, str2ints, fstint
 )
 from ._schema import (
     extract_record_type, acquaint_schema, populate_schema_defs,
     extract_logical_type
 )
 from ._reader_common import (
-    SchemaResolutionError, MAGIC, SYNC_SIZE, HEADER_SCHEMA
+    ReadError, SchemaResolutionError, MAGIC, SYNC_SIZE, HEADER_SCHEMA
```

Comment:
How is this not throwing an `ImportError`? Wasn't it removed from `_reader_common`?

Reply:
Ah, this was tricky! It actually *was* throwing an `ImportError`, but in `reader.py`, we catch `ImportError` and fall back to the pure Python implementation, which masks the error. This is nasty. I am adding an automated test to catch this: it will verify that in CPython, we are loading the Cython module, and in Pypy, we are loading the `.py` version.

```

```

Reply:
Ah, that explains it. And thanks for the test.